src/js/controllers/modals/wallets.js

```diff
@@ -2,25 +2,24 @@
 
 angular.module('copayApp.controllers').controller('walletsController', function($scope, bwsError, profileService) {
 
-  var self = $scope.self;
-
   $scope.selectWallet = function(walletId, walletName) {
-    if (!profileService.getClient(walletId).isComplete()) {
-      self.error = bwsError.msg({
-        'code': 'WALLET_NOT_COMPLETE'
-      }, 'Could not choose the wallet');
-      self.error = {
-        errors: [{
-          message: 'The Wallet could not be selected'
-        }]
-      };
-      $scope.cancel();
-      return;
-    }
-    self.selectedWalletId = walletId;
-    self.selectedWalletName = walletName;
-    self.fc = profileService.getClient(self.selectedWalletId);
-    $scope.cancel();
+
+    var client = profileService.getClient(walletId);
+    $scope.errorSelectedWallet = {};
+
+    profileService.isReady(client, function(err) {
+      if (err) { 
+        $scope.errorSelectedWallet[walletId] = bwsError.msg(err);
+        return;
+      }
+
+      var obj = {
+        'walletId': walletId,
+        'walletName': walletName,
+        'client': profileService.getClient(walletId)
```

Comment:
no need to send the client or the wallet name.  The called can call profileService and get that info. Better only return the walletId.